### PR TITLE
pyGPCCA integration

### LINKED ISSUE #168: Tests for msmtools/GPCCA
These would be very nice - it would reduce my fear breaking everything... See https://github.com/msmdev/msmtools/issues/4

### LINKED ISSUE #469: Missing jax installation and stochastic mode
@Marius1311 I remember that an error popped up when trying `mode='stochastic'`, but jax has not been installed and that
this error was rather ominous.
This must've changed  in 1.1.0, when I've changed the detection (i.e. `'stochastic'` mode does not "need" jax, it just needs an attribute called `hessian` ([here](https://github.com/theislab/cellrank/blob/master/cellrank/tl/kernels/_velocity_kernel.py#L217)'s the check).
I did add dummy functions [here](https://github.com/theislab/cellrank/blob/master/cellrank/tl/kernels/_velocity_schemes.py#L73) which should produce a nice error message, but they don't.

#### Versions:
cellrank==1.1.0

### PR DESCRIPTION
**Pull request by michalk8**
**IMPORTANT: Please search among the [Pull request](../pulls) before creating one.**

## Title

Adds pyGPCCA as a dependency.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)

## Description

Also updates pre-commit and fixes some docstrings, fixes #469  and fixes `nbviewer` CSS.

## Closes

closes #168 #453 #465 #469

TODOs:
- [x] verify that we're reading the correct attributes
- [x] search for and remove all previous `msmtools` things
- [ ] ~~in `compute_schur`, we say which clusters not to pick, maybe it would be better to say otherwise~~ I think this can be too much "overwhelming" info (i.e. the set of states we can't select is small than those we can), so I'd keep it as it is.
- [x] @Marius1311 discuss how to handle this change: (now) sparse matrix + brands = error; (before) ... = densify; currently, we issue a warning and densify (I think it's ok), alt. is to have a e.g. `cr.settings.densify = False`, which prevents doing this

### CODE REVIEW

**Comment by Marius1311** [comment]
Hi @michalk8, re the last point, I would vote for warning and densifying. The reason is that this case will be **very** common: our matrices are basically always sparse, and many people, at least when initially installing, will not have SLEPSc, hence they are in exactly the situation you described.

File: cellrank/datasets/_datasets.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
```
Comment (Marius1311) [inline cellrank/datasets/_datasets.py:1]: Maybe just tell me once why you removed all of these?

File: cellrank/pl/_graph.py [hunk 2]
```diff
@@ -174,7 +173,7 @@ def plot_arrows(curves, G, pos, ax, edge_weight_scale):
                 X, Y = X[::-1], Y[::-1]
 
             mid = len(X) // 2
-            posA, posB = zip(X[mid : mid + 2], Y[mid : mid + 2])  # noqa
```
Comment (Marius1311) [inline cellrank/pl/_graph.py:177]: What does the # noqa do?

File: cellrank/tl/kernels/_velocity_kernel.py [hunk 3]
```diff
@@ -214,13 +213,19 @@ def compute_transition_matrix(
...
+                    scheme, "hessian"
+                ):  # this can also raise in our velocity scheme definition
+                    raise NotImplementedError
+            except NotImplementedError:
+                logg.warning(
+                    f"Unable to detect a method for Hessian computation. If using predefined functions, consider "
+                    f"installing `jax` as `pip install jax jaxlib`\n"
+                    f"Defaulting to `mode={VelocityMode.MONTE_CARLO.s!r}` and `n_samples={n_samples}`"
+                )
+                mode = VelocityMode.MONTE_CARLO
```
Comment (Marius1311) [inline cellrank/tl/kernels/_velocity_kernel.py:228]: I guess this fixed the Jax error I was experiencing?

**Review by Marius1311 — APPROVED** [review]
Mike, this looks very well done. Did you have any troubles with tests or did they pass straight away?

File: cellrank/pl/_graph.py (hunk 2 above)
Comment (michalk8) [reply cellrank/pl/_graph.py:177]: It's used to silence flake8

File: cellrank/datasets/_datasets.py (hunk 1 above)
Comment (michalk8) [reply cellrank/datasets/_datasets.py:1]: They were only supposed to be used for Python2.

File: cellrank/tl/kernels/_velocity_kernel.py (hunk 3 above)
Comment (michalk8) [reply cellrank/tl/kernels/_velocity_kernel.py:228]: Yes.

**Comment by michalk8** [comment]
Thanks for the review, merging.

File: cellrank/datasets/_datasets.py (hunk 1 above)
Comment (Marius1311) [reply cellrank/datasets/_datasets.py:1]: ok, got it.

File: cellrank/pl/_graph.py (hunk 2 above)
Comment (Marius1311) [reply cellrank/pl/_graph.py:177]: Okay, that's what I was guessing. I suppose the new tags silence only particular flake8 complaints.

File: cellrank/tl/kernels/_velocity_kernel.py (hunk 3 above)
Comment (Marius1311) [reply cellrank/tl/kernels/_velocity_kernel.py:228]: amazing, thanks.

**Comment by Marius1311** [comment]
nice!

**Pull request merged**